src/TestFramework/TestFramework/Interfaces/EmptyDataSourceExceptionInfoExtensions.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Globalization;
+using System.Reflection;
+
+namespace Microsoft.VisualStudio.TestTools.UnitTesting;
+
+internal static class EmptyDataSourceExceptionInfoExtensions
+{
+    internal static ArgumentException GetExceptionForEmptyDataSource(this ITestDataSource dataSource, MethodInfo testMethodInfo)
+        => dataSource is ITestDataSourceEmptyDataSourceExceptionInfo info ? info.GetExceptionForEmptyDataSource(testMethodInfo) : new ArgumentException(
+            string.Format(
+                CultureInfo.InvariantCulture,
+                FrameworkMessages.DynamicDataIEnumerableEmpty,
+                "GetData",
+                dataSource.GetType().Name));
```

Comment:
Not mandatory but I tend to find it easier when scanning code to have the ternary split like

```suggestion
        => dataSource is ITestDataSourceEmptyDataSourceExceptionInfo info
            ? info.GetExceptionForEmptyDataSource(testMethodInfo)
            : new ArgumentException(
                string.Format(
                    CultureInfo.InvariantCulture,
                    FrameworkMessages.DynamicDataIEnumerableEmpty,
                    "GetData",
                    dataSource.GetType().Name));
```